src/frontend/src/handler/create_table.rs

```diff
@@ -1367,6 +1369,21 @@ pub async fn handle_create_table(
         risingwave_sqlparser::ast::Engine::Hummock => Engine::Hummock,
         risingwave_sqlparser::ast::Engine::Iceberg => Engine::Iceberg,
     };
+    if engine == Engine::Iceberg && handler_args.with_options.get_connector().is_some() {
+        // HACK: since we don't have atomic DDL, table with connector may lose data.
+        // FIXME: remove this after https://github.com/risingwavelabs/risingwave/issues/21863
+        if let Some(rate_limit) = handler_args.with_options.insert(
+            OverwriteOptions::SOURCE_RATE_LIMIT_KEY.to_owned(),
+            "0".to_owned(),
+        ) {
+            // prevent user specified rate limit
+            return Err(ErrorCode::NotSupported(
+                "source_rate_limit for iceberg table engine during table creation".to_owned(),
+                "Please remove source_rate_limit from WITH options.".to_owned(),
+            )
+            .into());
+        }
+    }
```

Comment:
We can put this logic inside the function `create_iceberg_engine_table`.

Reply:
Not actually, when we enter `create_iceberg_engine_table`, the planning for table with connector (`bind_create_source_or_table_with_connector`) is already finished. 🤪